Drag.py

```diff
@@ -0,0 +1,189 @@
+#!/usr/bin/env python
+import rospy
+from geometry_msgs.msg import WrenchStamped
+from nav_msgs.msg import Odometry
+# Initialized ros parameters and global values (feel free to tweak, values were empirically gained through trial/error)
+rospy.set_param('LINEAR_FORCE', 10)
+rospy.set_param("TORQUE", 5)
+rospy.set_param('ForceDown', -20)
+rospy.set_param('TimeOfForceDown', 10)
+Linear_Drag_X = 0
+Linear_Drag_Y = 0
+Linear_Drag_Z = 0
+Roll_Drag = 0
+Pitch_Drag = 0
+Yaw_Drag = 0
+Velocity = 0
+'''
+Velocities between two time intervals will never be truly 'equal' in real world
+Must compare the abs of the difference of the two with a small delta value
+if the difference is less than this value we can conclude that the numbers are satisfyingly close
+also used as a value that is satisfyingly close to zero. Hard-coded (can be changed).
+'''
+delta = .00001
+# Max velocity cannot be initialized to 0 or there will be an initial divide-by-zero error
+Max_Velocity = .1
+Bouyancy = 0
+# Magnitude of applied force for determining drag coeffcients in linear axes
+appliedLinearForce = rospy.get_param("LINEAR_FORCE", default=10)
+# Magnitude of applied torque for determining drag coeffcients in rotational axes
+appliedTorque = rospy.get_param("TORQUE", default=5)
+appliedForceDown = rospy.get_param('ForceDown', default=-20)
+TimeOfAplliedForceDown = rospy.get_param('TimeOfForceDown')
+
+
+def Find_Bouyancy(choice):  # Initial function used to upward force of buoyancy (used in determining drag in Z axis)
+    global Bouyancy
+    Down_Force = -.5  # Applies initial downward force in z axis
+    pub = rospy.Publisher('/wrench', WrenchStamped, queue_size=30)
+    rospy.init_node('move_sub', anonymous=False)
+    force_msg = WrenchStamped()
+    force_msg.wrench.force.z = appliedForceDown
+    pub.publish(force_msg)  # publish wrench with force
+    rospy.sleep(TimeOfAplliedForceDown)  # node sleeps for some amount of time before continuing
+    force_msg.wrench.force.z = 0  # Applies 0 force which stops downward force
+    pub.publish(force_msg)
+    while not (rospy.is_shutdown()):
+        rospy.Subscriber("/odom", Odometry, get_velocity, choice)
+        rospy.sleep(1)
+        if Velocity > 0.0:
+            while not (rospy.is_shutdown()):
+                force_msg.wrench.force.z = Down_Force
+                pub.publish(force_msg)
+                Down_Force = Down_Force - .001
+                rospy.sleep(.01)
+                if Velocity < 0.0:
```

Comment:
Remember that the sub's velocity is estimated from sensors, so it will be noisy. Even if the sub is perfectly still, you'll observe a velocity bouncing around zero.

One might expect this to be compounded by the fact bobbing of the sub in the water